### PR TITLE
bpo-40275: Add os_helper submodule in test.support

### PR DESCRIPTION
**Pull request by shihai1991**
remove the following vars, functions, classes in os_helper:
* rmtree()
* unlink()
* rmdir()
* FS_NONASCII
* TESTFN_UNICODE, TESTFN_NONASCII, TESTFN_UNENCODABLE, TESTFN_UNDECODABLE
* SAVEDCWD 
* temp_dir()
* change_cwd()
* temp_cwd()
* temp_umask()
* create_empty_file()
* make_bad_fd()
* EnvironmentVarGuard
* can_symlink(), skip_unless_symlink()
* can_xattr(), skip_unless_xattr()
* fs_is_case_insensitive()
* fd_count()
* FakePath

https://bugs.python.org/issue40275

### CODE REVIEW

File: Lib/test/support/os_helper.py [hunk 1]
```diff
@@ -0,0 +1,623 @@
...
+    create or change the CWD, an error is raised.  If *quiet* is True,
+    only a warning is raised and the original CWD is used.
+
+    """
+    with temp_dir(path=name, quiet=quiet) as temp_path:
+        with change_cwd(temp_path, quiet=quiet) as cwd_dir:
+            yield cwd_dir
+
+
+def findfile(filename, subdir=None):
```
Comment (vstinner) [inline Lib/test/support/os_helper.py]: Hum, findfile() is not **directly** related to the os module. It looks more like a generic helper for tests. Please leave this one in test.support.

File: Lib/test/support/os_helper.py [hunk 2]
```diff
@@ -0,0 +1,623 @@
...
+    return filename
+
+
+def create_empty_file(filename):
+    """Create an empty file. If the file already exists, truncate it."""
+    fd = os.open(filename, os.O_WRONLY | os.O_CREAT | os.O_TRUNC)
+    os.close(fd)
+
+
+def skip_unless_symlink(test):
```
Comment (vstinner) [inline Lib/test/support/os_helper.py]: Please move it just after can_symlink() definition.

File: Lib/test/support/os_helper.py [hunk 3]
```diff
@@ -0,0 +1,623 @@
...
+    if dir_created:
+        pid = os.getpid()
+    try:
+        yield path
+    finally:
+        # In case the process forks, let only the parent remove the
+        # directory. The child has a different process id. (bpo-30028)
+        if dir_created and pid == os.getpid():
+            rmtree(path)
+
```
Comment (vstinner) [inline Lib/test/support/os_helper.py:401]: PEP 8: please add an empty line.

Comment (shihai1991) [reply Lib/test/support/os_helper.py:401]: Get it. I use flake8 to fix all pep8 warnings. Make sure all new python file have no pep8 problems.

File: Lib/test/support/os_helper.py (hunk 2 above)
Comment (shihai1991) [reply Lib/test/support/os_helper.py]: done, thanks.

File: Lib/test/support/os_helper.py (hunk 1 above)
Comment (shihai1991) [reply Lib/test/support/os_helper.py]: make sense.

File: Lib/test/support/__init__.py [hunk 4]
```diff
@@ -22,8 +21,34 @@
 import unittest
 import warnings
 
+from .os_helper import FS_NONASCII
```
Comment (vstinner) [inline Lib/test/support/__init__.py]: Can you please group these imports? from .os_helper import (FS_NONASCII, SAVEDCWD, ...)?

Comment (shihai1991) [reply Lib/test/support/__init__.py]: updated.

**Pull request merged**